modules/service/src/main/resources/db/migration/V0030__observations.sql

```diff
@@ -171,6 +171,22 @@ create table t_observation (
   c_hour_angle_min     d_hour_angle        null        default null,
   c_hour_angle_max     d_hour_angle        null        default null,
 
+  -- a column we can use to identify and join with distinct observing conditions in this program
+  c_conditions_key     text not null generated always as (
+    immutable_concat_ws(
+      ',',
+      c_program_id,
+      c_cloud_extinction,
+      c_image_quality,
+      c_sky_background,
+      c_water_vapor,
+      coalesce(c_air_mass_min::text, 'null'),
+      coalesce(c_air_mass_max::text, 'null'),
+      coalesce(c_hour_angle_min::text, 'null'),
+      coalesce(c_hour_angle_max::text, 'null')
+    )
+   ) stored,
+
```

Comment:
To group observations by conditions I concatenate all the relevant fields together and use that as the key. I need to do this so the constraint set group has a key in the obs table that it can join with.